packages/resolvers/default/src/DefaultResolver.js

```diff
@@ -179,6 +194,20 @@ class NodeResolver {
     return resolved;
   }
 
+  shouldIncludeNodeModule(env, name) {
+    if (env.includeNodeModules === false) {
+      return false;
+    }
+
+    if (Array.isArray(env.includeNodeModules)) {
+      let parts = this.getModuleParts(name);
+      // $FlowFixMe - flow is dumb
```

Comment:
Because the call to `getModuleParts` invalidates the check, I suggest:
```js
  shouldIncludeNodeModule({includeNodeModules}, name) {
    if (includeNodeModules === false) {
      return false;
    }

    if (Array.isArray(includeNodeModules)) {
      let parts = this.getModuleParts(name);
      return includeNodeModules.includes(parts[0]);
    }

    return true;
  }
```